prov/efa/src/efa_hmem.c

```diff
@@ -407,44 +331,27 @@ int efa_domain_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem
  * @return  0 on success
  *          negative libfabric error code on an unexpected error
  */
-int efa_domain_hmem_info_init_all(struct efa_domain *efa_domain)
+int efa_domain_hmem_support_init_all(struct efa_domain *efa_domain)
 {
-	int ret, err;
+	int ret = 0;
+	enum fi_hmem_iface ifaces[4] = {FI_HMEM_SYSTEM, FI_HMEM_CUDA,
+	                                FI_HMEM_NEURON, FI_HMEM_SYNAPSEAI};
 
 	if(g_device_cnt <= 0) {
 		return -FI_ENODEV;
 	}
 
 	memset(efa_domain->hmem_info, 0, OFI_HMEM_MAX * sizeof(struct efa_hmem_info));
 
-	ret = 0;
-
-	err = efa_domain_hmem_info_init_system(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the System hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_cuda(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Cuda hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_neuron(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Neuron hmem_info struct! err: %d\n",
-			 err);
-	}
-
-	err = efa_domain_hmem_info_init_synapseai(efa_domain);
-	if (err) {
-		ret = err;
-		EFA_WARN(FI_LOG_DOMAIN, "Failed to populate the Synapseai hmem_info struct! err: %d\n",
-			 err);
+	for (int i = 0; i < sizeof(ifaces) / sizeof(enum fi_hmem_iface); ++i) {
+		ret = efa_domain_hmem_info_init_iface(efa_domain, ifaces[i]);
+		if (ret) {
+			EFA_WARN(FI_LOG_DOMAIN,
```

Comment:
similar comment as above, if `efa_domain_hmem_info_init_iface` doesn't return non-zero, we don't have to check return status.

Reply:
Addressed